src/float_cast.h

```diff
@@ -106,17 +106,40 @@
 
 		return intgr ;
 	}
-#elif (defined (WIN32) || defined (_WIN32)) && defined(_M_3X64)
+#elif (defined (WIN32) || defined (_WIN32)) && defined(_M_X64)
```

Comment:
I can't find the documentation about symbols _M_3X64 and _M_X64.  Are you sure this doesn't disable the optimizations in 32 bit builds?  Maybe we don't care if it does.

Reply:
_M_3X64 was an (intentional) type, I was surprised by it as well.

_M_X64 is defined by MSVC compiler: https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-160

That means, that we are missing MinGW/Cygwin, but this shouldn't be a problem, as GCC is assumed to have an adequate implementation for lrint family.

In any case, for 32 bit MSVC builds the first branch will be chosen. Probably, we can improve the performance using SSE as well. We have an SSE2 requirement, but it is used only for the libsbsms and soundtouch, not by Audacity itself (which I find surprising).

Reply:
Surprisingly, 64-bit executable now is twice as fast when exporting WAV. I wasn't able to achieve a similar result while playing with 32 bit executable, but the SSE version was still a bit faster

Reply:
First branch, I get it.

Several years ago wrote src/effects/Equalization48x.* but they are EXPERIMENTALs we don't compile and I think they no longer do compile.

I'm not well versed in this topic of streaming instructions and never dug deeply into how that file was supposed to work.  But there are also other things in EQ (the actually used Equalization.*) that maybe should not be in there, but be more widely available as utilities for other built-in effects.

(The individual who wrote Equalization48x was also following my rewrite of noise reduction on the forum, and made some suggestions about algorithms there... which weren't good.  But I didn't presume that this other work wasn't also good.)